src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java

```diff
@@ -77,11 +78,37 @@ public DefaultCaseDefinitionRepository(final ApplicationParams applicationParams
         }
     }
 
+    @Override
+    public List<String> getCaseTypesReferences() {
+        return getCaseTypesReferences(null);
+    }
+
+    public List<String> getCaseTypesReferences(final HttpHeaders httpHeaders) {
+        LOG.debug("retrieving case type references");
+        try {
+            final HttpEntity requestEntity = new HttpEntity<CaseType>(httpHeaders != null ? httpHeaders : securityUtils.authorizationHeaders());
+            return Arrays.asList(restTemplate.exchange(applicationParams.caseTypesReferencesDefURL(), HttpMethod.GET, requestEntity, String[].class).getBody());
+
+        } catch (Exception e) {
+            LOG.warn("Error while retrieving case type references", e);
+            if (e instanceof HttpClientErrorException
+                && ((HttpClientErrorException)e).getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException("Resource not found when getting case type references because of " + e.getMessage());
+            } else {
+                throw new ServiceException("Problem getting case type references because of " + e.getMessage());
+            }
+        }
+    }
+
     @Override
     public CaseType getCaseType(final String caseTypeId) {
+        return getCaseType(caseTypeId, null);
+    }
+
+    public CaseType getCaseType(final String caseTypeId, final HttpHeaders httpHeaders) {
```

Comment:
Can we declare httpHeaders Optional? rather than passing a null?

Reply:
Optional as method parameters? It is advised to use it as a return type only. Please see:
http://dolszewski.com/java/java-optional-parameters/
3.3. Why not Optional as method argument?